src/protocolgame.cpp

```diff
@@ -2289,7 +2289,8 @@ void ProtocolGame::sendCreatureHealth(const Creature* creature)
 	if (creature->isHealthHidden()) {
 		msg.addByte(0x00);
 	} else {
-		msg.addByte(std::ceil((static_cast<double>(creature->getHealth()) / std::max<int32_t>(creature->getMaxHealth(), 1)) * 100));
+		int32_t maxHealth = std::max(creature->getMaxHealth(), 1);
+		msg.addByte(std::ceil((static_cast<double>(std::min(creature->getHealth(), maxHealth)) / maxHealth) * 100));
```

Comment:
This will only fix client side (so showing it in-game) server still gonna hold invalid values I guess?

Reply:
I am not 100% sure, but as I understand it, this information comes from the server to the client, and this calculation should not be done at any time from the client side, so if everything is fine from the server, it should not be wrong in the tibia-client, if I'm wrong please notify me.

Reply:
yeah, but value in server gonna still be broken like max health being for example 100 while health is 200. I don't think this should happen server side, so the fix should be in server sources not at the time when they are sent to client.

Reply:
and/or at least add a console warning about the mismatching values

Reply:
> yeah, but value in server gonna still be broken like max health being for example 100 while health is 200. I don't think this should happen server side, so the fix should be in server sources not at the time when its sent to client.

I understand what you are saying, just add this in case something weird happens (example: force current health from sources.), But I have not been able to find anything that can affect amounts of current health and maximum health, cannot be broken with luascripts , and neither with equipment methods.

Reply:
I will leave my opinion about it

You shouldn't want to suppress the side effects of something wrong. You will be hiding that there is something wrong. If something is wrong, it must be visible for someone to see and correct it.

You want to correct what is wrong. In this case, the best thing to do is to alert the user (with warning) that healthNow is greater than healthMax (validation of monsters.xml?) (when the monster/creature is instantiated?)

Fix the cause, not the symptom!